nautobot_chatops_ipfabric/ipfabric.py

```diff
@@ -116,6 +129,38 @@ def get_path_simulation(
         payload = {}
         return self.get_response_json("GET", "/api/v1/graph/end-to-end-path", payload, params)
 
+    def get_pathlookup(
+        self, src_ip, dst_ip, src_port, dst_port, protocol, snapshot_id
+    ):  # pylint: disable=too-many-arguments
+        """Return pathlookup simulation as PNG output. Requires v4 IP Fabric server."""
+        payload = {
+            "snapshot": snapshot_id,
+            "parameters": {
+                "type": "pathLookup",
+                "pathLookupType": "unicast",
+                "protocol": protocol,
+                "startingPoint": src_ip,
+                "startingPort": src_port,
+                "destinationPoint": dst_ip,
+                "destinationPort": dst_port,
+                "groupBy": "siteName",
+                "networkMode": "true",
+                "securedPath": "false",
+            },
+        }
+        logger.debug(  # pylint: disable=logging-too-many-args
+            "Received end-to-end PNG path simulation request: ", payload
+        )
+
+        # no params required
+        params = {}
+        response = self.get_response_raw("POST", "/api/v1/graphs/png", payload, params=params)
+
+        # IP Fabric can potentially return a blank PNG file with status_code 200 when no path exists
+        if int(response.headers.get("Content-Length")) < 200:
```

Comment:
something not clear here, you talk about status_code, but you compare to content-lenght 🤔

Reply:
I'm going to change this. Very flaky solution. From talking to the IP Fabric guys, if I get the JSON data and check the `flag` attribute, it will give a more accurate indication of a fully composed PNG file.